renku_notebooks/util/kubernetes_.py

```diff
@@ -299,3 +300,77 @@ def _secret_exists(name, namespace):
     except client.rest.ApiException:
         pass
     return False
+
+
+def create_pvc(
+    name,
+    username,
+    git_namespace,
+    project_id,
+    commit_sha,
+    git_host,
+    storage_size,
+    storage_class="default",
+):
+    """Create a PVC."""
+
+    # check if we already have this PVC
+    pvc = _get_pvc(username, project_id, commit_sha)
+
+    if pvc:
+        status = "existing"
+    if not pvc:
+        pvc = client.V1PersistentVolumeClaim(
+            metadata=client.V1ObjectMeta(
```

Comment:
Can you add the branch name here? It would be nice to have all arguments of `make_server_name(namespace, project, branch, commit_sha)` easily available in order to identify a user server (or create the UserServer object once that refactoring is through) from the info that is easily available without any further API requests.

Reply:
I could just use `make_server_name` for the pvc name?

Reply:
@rokroskar ☝️ I think using `make_server_name` in the pvc name is a great idea. See my other comment about `_get_pvc` for my reasoning why.